todotxt: Portfile fixes to bash_completion variant and config file regex

woutgg opened the pull request:
#### Description

- The bash_completion script was installed unconditionally, now it will only be installed if the variant is actually enabled.
- The regex patching 'todo.cfg' to include a different `TODO_DIR` was outdated and has been fixed.

###### Type(s)
- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
macOS 10.12.6 16G1408
Xcode 8.3.3 8E3004b

###### Verification
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint`?
- [x] tried a full install with `sudo port -vst install`?

pmetzger commented:
I think there's no reason to have variants for bash completion scripts. They're tiny, and it makes more sense to me to just install them unconditionally, which is what is generally done. @nerdling ?

pmetzger commented on office/todotxt/Portfile:
```diff
@@ -38,15 +38,17 @@ destroot {
+        set completions_path ${destroot}${prefix}/share/bash-completion/completions
+        xinstall -d ${completions_path}
+        xinstall -m 644 ${worksrcpath}/todo_completion ${completions_path}/${name}
+    }
```
Okay, so, as I suggested in the other comments, I don't think we need a variant for this, and I'd suggest you reduce this PR to just fixing the reinplace.

woutgg commented:
Updated: I have removed the variant and also the bash-completion dependency, in order not to force non-bash users to have bash + completion support installed.

pmetzger commented:
@woutgg Cool. You should fix the commit message, though. I'll then merge the change.

woutgg commented:
Done, I hope this is better.

pmetzger commented:
Merged!
